### PR TITLE
feat: list public codemods

### PR DESCRIPTION
**Pull request by henoktse**

### CODE REVIEW

File: intuita-webview/src/shared/Collapsable/Collapsable.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+						})}
+						onClick={() => setExpanded(!expanded)}
+					>
+						<ArrowDownIcon
+							className={cn(
+								'collapsable__arrow',
+								headerChevronClassName,
+								{
+									'collapsable__arrow--collapsed': !expanded,
+								},
```
Comment (hbjORbj) [inline intuita-webview/src/shared/Collapsable/Collapsable.tsx:73]:
```suggestion
								!expanded && 'collapsable__arrow--collapsed',
```

File: src/components/webview/CodemodListProvider.ts [hunk 2]
```diff
@@ -87,7 +90,8 @@ export class CodemodListPanelProvider implements WebviewViewProvider {
 		this.__view = webviewView;
 
 		this.__view.onDidChangeVisibility(() => {
-			this.getCodemodTree();
+			this.getCodemodTree('recommended');
+			this.getCodemodTree('public');
```
Comment (hbjORbj) [inline src/components/webview/CodemodListProvider.ts:94]: Maybe `enum` for these string values?

File: src/components/webview/CodemodListProvider.ts [hunk 3]
```diff
@@ -133,23 +137,45 @@ export class CodemodListPanelProvider implements WebviewViewProvider {
...
-			viewId: 'codemodList',
-			viewProps: {
-				data: treeNodes?.[0],
-			},
-		});
+			if (recommended) {
+				return this.setView({
+					viewId: 'codemodList',
+					viewProps: {
+						data: treeNodes?.[0],
```
Comment (hbjORbj) [inline src/components/webview/CodemodListProvider.ts:161]:
```suggestion
						data: treeNodes[0],
```

File: src/components/webview/CodemodListProvider.ts [hunk 4]
```diff
@@ -133,23 +137,45 @@ export class CodemodListPanelProvider implements WebviewViewProvider {
...
+		try {
+			if (recommended) {
+				await this.__codemodService.getCodemods();
+			} else {
+				await this.__codemodService.getDiscoverdCodemods();
+			}
+			const codemodList = this.__getCodemod(recommended);
+			const treeNodes = codemodList.map((codemod) =>
+				this.__getTreeNode(codemod),
+			);
```
Comment (hbjORbj) [inline src/components/webview/CodemodListProvider.ts:155]: let's validate treeNodes here

File: src/components/webview/CodemodListProvider.ts [hunk 5]
```diff
@@ -133,23 +137,45 @@ export class CodemodListPanelProvider implements WebviewViewProvider {
...
+				return this.setView({
+					viewId: 'codemodList',
+					viewProps: {
+						data: treeNodes?.[0],
+					},
+				});
+			}
+			this.__postMessage({
+				kind: 'webview.codemodlist.setPublicCodemodList',
+				value: treeNodes?.[0],
```
Comment (hbjORbj) [inline src/components/webview/CodemodListProvider.ts:167]:
```suggestion
				value: treeNodes[0],
```

**Review by hbjORbj — CHANGES_REQUESTED** [review]

**Comment by hbjORbj** [comment]
Can you adapt your code after rebasing on the main branch? No changes are shown in the file explorer view.

https://user-images.githubusercontent.com/32841130/234552402-d7475936-aff2-49fb-a85c-b7549dccf553.mov

File: src/components/webview/CodemodListProvider.ts (hunk 4 above)
Comment (hbjORbj) [reply src/components/webview/CodemodListProvider.ts:155]: Validate `treeNodes` here so you don't need to do `treeNodes?.[0]` below

File: src/components/webview/CodemodListProvider.ts (hunk 3 above)
Comment (hbjORbj) [reply src/components/webview/CodemodListProvider.ts:161]: We can validate `treeNodes` above so you don't need to do `treeNodes?.[0]`

File: intuita-webview/src/codemodList/App.tsx [hunk 6]
```diff
@@ -1,16 +1,22 @@
...
+import type { View, WebviewMessage, CodemodTreeNode } from '../shared/types';
 import TreeView from './TreeView';
+import { Container, LoadingContainer } from './components/Container';
+import { VSCodeProgressRing } from '@vscode/webview-ui-toolkit/react';
 
 type MainViews = Extract<View, { viewId: 'codemodList' }>;
 
 function App() {
 	const [view, setView] = useState<MainViews | null>(null);
+	const [publicCodemods, setPublicCodemods] =
```
Comment (grzpab) [inline intuita-webview/src/codemodList/App.tsx:13]: you can use either from `fp-ts` to have a state for correct data and the error

File: src/components/webview/CodemodListProvider.ts (hunk 2 above)
Comment (grzpab) [reply src/components/webview/CodemodListProvider.ts:99]: string literals are fine :)

File: src/components/webview/CodemodListProvider.ts [hunk 7]
```diff
@@ -133,23 +137,45 @@ export class CodemodListPanelProvider implements WebviewViewProvider {
...
 	};
 
-	public async getCodemodTree() {
-		await this.__codemodService.getCodemods();
-		const codemodList = this.__getCodemod();
-		const treeNodes = codemodList.map((codemod) =>
-			this.__getTreeNode(codemod),
-		);
+	public async getCodemodTree(type: 'recommended' | 'public') {
+		const recommended = type === 'recommended';
```
[the remark this answers is not in the document]
Comment (grzpab) [reply src/components/webview/CodemodListProvider.ts:151]: it's ok with me to keep the variable, but I wouldn't normally use it in my code :)

File: intuita-webview/src/shared/Collapsable/Collapsable.tsx (hunk 1 above)
Comment (grzpab) [reply intuita-webview/src/shared/Collapsable/Collapsable.tsx:73]: the first one is the one in accordance with `classnames` lib docs

File: src/components/webview/CodemodListProvider.ts (hunk 4 above)
Comment (grzpab) [reply src/components/webview/CodemodListProvider.ts:166]: the first tree node might not exist, but the view has to be created.

File: intuita-webview/src/shared/Collapsable/Collapsable.tsx (hunk 1 above)
Comment (hbjORbj) [reply intuita-webview/src/shared/Collapsable/Collapsable.tsx:73]: gottcha

**Review by hbjORbj — APPROVED** [review]

**Review by grzpab — APPROVED** [review]

**Pull request merged**